### PR TITLE
feat: add learning materials page

### LINKED ISSUE #1227: Feat: implement learning materials page on the community view page.
- [ ] Ask for the correct text description from Araya
- [x] Reuse the course and the existing learning module card.
- [x] Translate the description and the title
- [x] Section should be between "challenge" and the "scoreboard"
- [x] When clicked it should redirect to the learning content section.
- [x] Section should display something when we have empty content [Even though this is unlikely]
- [x] The section should show a loading state
- [x] course cards expands the full width of the row everywhere. note: pay attention to the max-width of the content text
- [x] module cards are divided into half of the width of the row. this visually indicates to the user that the modules are smaller than the course

Design : https://www.figma.com/file/0BRR6Ch2qbOEUT4zIdmmzg/Dacade-2?type=design&node-id=2063-22389&mode=design&t=l7UWddTYnmeAeoky-4

### PR DESCRIPTION
**Pull request by serapieTuyishime**
# Submit a pull request

- [x] This is not a duplicate of an existing pull request.
- [x] No existing features have been broken without good reason.
- [x] Your commit messages are detailed
- [x] The code style guideline have been followed.
- [x] Documentation has been updated to reflect your changes.
- [ ] Tests have been added or updated to reflect your changes.
- [ ] All tests pass.

---

Replace any ":question:" below with information about your pull request.

## Pull Request Details

This aims to create the new learning materials page on the community

:question:

## Breaking Changes

Describe what features are broken by this pull request and why, if any.

## Issues Fixed

Closes #1227

### CODE REVIEW

File: src/components/sections/communities/overview/Sidebar.tsx [hunk 1]
```diff
@@ -30,19 +30,36 @@ export default function Sidebar(): JSX.Element {
   };
 
   const scoreboardLink = hasCurrentCommunity ? `/communities/${currentCommunity.slug}/scoreboard` : "";
+  const learningMaterialsLink = hasCurrentCommunity ? `/communities/${currentCommunity.slug}/learning-materials` : "";
   const mainLink = hasCurrentCommunity ? `/communities/${currentCommunity.slug}` : "";
```
Comment (mhenriette) [inline src/components/sections/communities/overview/Sidebar.tsx:34]: consider optimizing these links by using a `useMemo` hook
```suggestion
  const links = useMemo(() => {
    if (!hasCurrentCommunity)
      return {
        scoreboardLink: "",
        learningMaterialsLink: "",
        mainLink: "",
      };
    const { slug } = currentCommunity;
    return {
      scoreboardLink: `/communities/${slug}/scoreboard`,
      learningMaterialsLink: `/communities/${slug}/learning-materials`,
      mainLink: `/communities/${slug}`,
    };
  }, [hasCurrentCommunity, currentCommunity]);
```

**Review by mhenriette — CHANGES_REQUESTED** [review]

**Review by mhenriette — COMMENTED** [review]
the pages looks good but I left few comments , consider implementing the reviews from @coderabbit as well.

The level label is missing
The course duration is missing 

also the alignment in the box is not consistent 
Screenshot 2024-06-04 at 4 47 08 PM

**Review by Samantha-KY — APPROVED** [review]

**Review by mhenriette — APPROVED** [review]

File: src/pages/communities/[slug]/index.tsx [hunk 2]
```diff
@@ -22,18 +24,27 @@ export default function Slug(props: {
...
-        <ChallengeCard key={challenge.id} data={challenge} community={community} />
-      ))}
-      <div className="md:hidden">
-        <div className="active md:hidden mb-7 scroll-smooth pt-5">
-          <div className="font-medium text-.5xl leading-snug">{t("communities.overview.scoreboard.title")}</div>
-          <div className="text-sm font-light lg:w-full lg:pr-7 pt-2">{t("communities.overview.scoreboard.description")} </div>
+    <>
+      <Head>
+        <title>{getMetadataTitle(t("communities.navigation.courses"), community?.name as string)}</title>
+        {getMetadataDescription(community?.description as string).map((attributes) => (
```
Comment (musayann) [inline src/pages/communities/[slug]/index.tsx:30]: can you use the component the metadata component here

File: src/pages/communities/[slug]/index.tsx [hunk 3]
```diff
@@ -22,18 +24,27 @@ export default function Slug(props: {
...
-      {challenges.map((challenge) => (
-        <ChallengeCard key={challenge.id} data={challenge} community={community} />
-      ))}
-      <div className="md:hidden">
-        <div className="active md:hidden mb-7 scroll-smooth pt-5">
-          <div className="font-medium text-.5xl leading-snug">{t("communities.overview.scoreboard.title")}</div>
-          <div className="text-sm font-light lg:w-full lg:pr-7 pt-2">{t("communities.overview.scoreboard.description")} </div>
+    <>
+      <Head>
+        <title>{getMetadataTitle(t("communities.navigation.courses"), community?.name as string)}</title>
```
Comment (musayann) [inline src/pages/communities/[slug]/index.tsx:29]: can you make the page title as a component which will then handle everything within itself

**Review by musayann — APPROVED** [review]

**Review by igorntwari — COMMENTED** [review]
**Q and A**

- [x] The challenge text is quite small compared to what we have on the Figma design 
- [x] The leaning material title is smaller than what we have on the Figma design 
- [x] The score-board title is bigger than what we have on the Figma design 
- [x] The responsiveness of the challenge button is breaking on medium screen   
image
- [ ] The gap between the button description and the arrow icon is quite big on medium screens and smartphones as well 
image
- [ ] The Submissions and Feedback on the small screens are flexed rows but on the design, they are flexed on the row 
image

**Review by musayann — APPROVED** [review]

**Pull request merged**